department-of-veterans-affairs/abd-vro
# EP Merge: Cancel EP400 after contentions have been migrated

## What was the problem?

Once the contentions have been merged, the contention needs to be cancelled. 
- Set the cancellation reason to:
   - Issues moved into or confirmed in pending EP[x] - claim #[y]
   - where [x] is the EP code 
   - [y] is the pending benefit claim ID

Associated tickets or Slack threads:
- #2056 

## How does this fix it?[^1]

Adds an additional hoppy client for the `getClaimDetailsQueue` in order to get the `endProductCode` from the claim details
Sets the cancellation reason to value above
Adds additional unit / integration tests for the added hoppy endpoint

## How to test this PR
- pull code
- from `/domain-ee/ee-ep-merge-app` run `pytest`
- from `/domain-ee/ee-ep-merge-app` run `pytest ./integration`

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
